mailpoet/mailpoet
# Allow multiple values in "is in country" segment [MAILPOET-3952]

This PR allows the selection of multiple values for the "is in country" segment and adds the operator None and Any.
It also changes the queries to use WC Lookup tables.

The tests have been modified to reflect the changes and there is a new integration test for the segment.
Because WooCommerce is not present for the integration tests, the PR also includes the creation of the lookup tables for the integration tests. The DB schema has been [copied from WC-Admin](https://github.com/woocommerce/woocommerce-admin/blob/49bc93f7e69d44c9da1cbd290a69092f6361002b/src/Install.php#L235).

To test:
* From the main branch, create a segment with a country.
* Switch to this branch
* Make sure the segment still works although the number of subscribers might be different if [this branch is not merged yet](https://github.com/mailpoet/mailpoet/compare/wc-guest-users).
* Edit the segment from MailPoet > Lists > Segments
* Check you can now select multiple countries
* Check it works with the None or Any operator
* Save the segment
* Check you can see the subscribers on the segment
* Repeat with a new segment
* Check that you can combine "is in country" with other filters.

New test:
`./do --test "test:integration --file=tests/integration/Segments/DynamicSegments/Filters/WooCommerceCountryTest.php"
`

[MAILPOET-3952]
